File: e2ee_sync_back/models/message.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// Message represents a message in the system
type Message struct {
	ID               uuid.UUID `json:"id"`
	UserID           uuid.UUID `json:"user_id"`
	EncryptedContent string    `json:"encrypted_content"`
	Nonce            string    `json:"nonce"`
	CreatedAt        time.Time `json:"created_at"`
}

File: cse_sync_front/src/features/dashboard/types/message.ts
export interface Message {
  id: string;
  user_id: string;
  content: string;
  created_at: string;
}

export interface EncryptedMessage {
  id: string;
  user_id: string;
  encrypted_content: string;
  nonce: string;
  created_at: string;
}

export interface CreateMessageRequest {
  content: string;
  nonce: string;
}

export type MessageSource = "network" | "cache";

export interface MessageFetchResult {
  messages: Message[];
  source: MessageSource;
}
